lib/carrierwave_direct/form_builder.rb

```diff
@@ -5,19 +5,21 @@ class FormBuilder < ActionView::Helpers::FormBuilder
     def file_field(method, options = {})
       @object.policy(enforce_utf8: true)
 
-      options.merge!(:name => "file")
-
-      fields = required_base_fields
-
-      fields << content_type_field(options)
-
-      fields << success_action_field(options)
+      fields = hidden_fields(options)
 
       # The file field must be the last element in the form.
       # Any element after this will be ignored by Amazon.
+      options.merge!(:name => "file")
```

Comment:
Shouldn't this option be added before `options` is passed to other methods?

Reply:
No because those options concern only file field and we don't want (and don't need) to pass them to the hidden fields, notice that we overwrite Rails' form builder file_field method:

https://github.com/rails/rails/blob/375a4143cf5caeb6159b338be824903edfd62836/actionview/lib/action_view/helpers/form_helper.rb#L2166

Than we call:

    fields << super

To add actual file field at the end.

Reply:
ah, yes, That makes sense.